### PR TITLE
Add rules for '35 more Semgrep rules' blog post

### PR DESCRIPTION
**Pull request by mschwager**

### CODE REVIEW

File: yaml/github-actions/rubygems-publish-key.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      subcategory: [audit]
+      confidence: HIGH
+      likelihood: HIGH
+      impact: MEDIUM
+      technology: [rubygems, github-actions]
+      references:
+        - https://guides.rubygems.org/trusted-publishing/
+        - https://guides.rubygems.org/api-key-scopes/#usage-with-gem-cli
+        - https://github.com/rubygems/release-gem
+    pattern: GEM_HOST_API_KEY
```
Comment (GrosQuildu) [inline yaml/github-actions/rubygems-publish-key.yaml:20]: Should we match `uses: ruby/setup-ruby` to reduce false positives?

**Review by GrosQuildu — CHANGES_REQUESTED** [review]
Maybe we should adjust "confidence" of generic tls rules? They may hit a lot of testing and conditional code.

Terraform has to syntaxes: https://developer.hashicorp.com/terraform/language/syntax
We should make our rules work with both.

File: hcl/nomad/podman-tls-verify-disabled.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            config {
+              ...
+              auth {
+                ...
+              }
+              ...
+            }
+            ...
+          }
+      - pattern: tlsVerify = false
```
[the remark this answers is not in the document]
Comment (mschwager) [reply hcl/nomad/podman-tls-verify-disabled.yaml:32]: Looks like that wouldn't work:

- https://developer.hashicorp.com/nomad/docs/job-specification/hcl2/expressions#bool
- https://developer.hashicorp.com/terraform/language/expressions/types#type-conversion

File: hcl/nomad/tls-hostname-verification-disabled.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      - pattern-inside: |
+          tls {
+            ...
+            key_file = ...
+            ...
+          }
+      - pattern-either:
+          - patterns:
+              - pattern-inside: tls { ... }
+              - pattern: verify_server_hostname = false
```
[the remark this answers is not in the document]
Comment (mschwager) [reply hcl/nomad/tls-hostname-verification-disabled.yaml:40]: Same as above.

File: yaml/github-actions/rubygems-publish-key.yaml (hunk 1 above)
Comment (mschwager) [reply yaml/github-actions/rubygems-publish-key.yaml:20]: I think this is just restricted to `gem push`, so FPs should be low. My concern with limiting to `uses: ruby/setup-ruby` is that `GEM_HOST_API_KEY` could occur in many places in the GHA file. The `env` can be set globally, per-job, per-step, etc.

Comment (mschwager) [reply yaml/github-actions/rubygems-publish-key.yaml:20]: Dropping the `likelihood` to `LOW` though 👍

File: generic/mysql-insecure-sslmode.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      subcategory: [audit]
+      cwe: "CWE-319: Cleartext Transmission of Sensitive Information"
+      confidence: HIGH
+      likelihood: HIGH
+      impact: HIGH
+      technology: [mysql]
+      references:
+        - https://dev.mysql.com/doc/connector-net/en/connector-net-8-0-connection-options.html
+        - https://dev.mysql.com/doc/connector-j/en/connector-j-connp-props-security.html
+    pattern-either:
```
[the remark this answers is not in the document]
Comment (mschwager) [reply generic/mysql-insecure-sslmode.yaml:18]: Hmm, looks like the scheme is optional: https://dev.mysql.com/doc/refman/8.4/en/connecting-using-uri-or-key-value-pairs.html.

Also, TODO, we should catch `ssh-password` and `ssh-identity-pass` usage.

**Review by GrosQuildu — APPROVED** [review]

**Pull request merged**